rust-keybroker/keybroker-server/src/challenge.rs

```diff
@@ -70,27 +75,43 @@ impl Challenger {
         &mut self,
         key_id: &str,
         wrapping_key: &PublicWrappingKey,
+        mock_challenge: bool,
     ) -> Challenge {
         // All challenges are given random u32 identities
-        let mut challenge_id: u32 = rand::random();
+        let mut challenge_id: u32 = self.rng.gen();
 
         // Simple lightweight collision avoidance - probably not needed given u32 distribution space,
         // but it's easy to do. Also allows us to throw out zero if we get it.
         while challenge_id == 0 || self.challenge_table.contains_key(&challenge_id) {
-            challenge_id = rand::random();
+            challenge_id = self.rng.gen();
         }
 
         let challenge = Challenge {
             challenge_id,
             key_id: key_id.to_owned(),
             wrapping_key: wrapping_key.clone(),
-            // TODO: We should create a random nonce here. The use of this mock value allows the
-            // server to be tested with a hard-coded example attestation token during development.
-            challenge_value: CCA_EXAMPLE_TOKEN_NONCE.to_vec(),
+            challenge_value: if mock_challenge {
+                CCA_EXAMPLE_TOKEN_NONCE.to_vec()
+            } else {
+                let mut v: Vec<u8> = vec![0; 64];
+                self.rng.fill(&mut v[..]);
```

Comment:
One thing I remember discussing with @thomas-fossati here is that we could create the challenge out of a combination of the client's public key hash plus additional random data. However, it's not entirely clear whether this adds any additional security, so I don't see an urgent need to address that in this PR. We can raise an issue to track and maybe address later if we think it's worth doing.

Reply:
Yes, we can certainly update the challenge creation later on to take into account extra data !